graylog2-web-interface/src/views/components/searchbar/queryvalidation/WarmTierErrorMessage.tsx

```diff
@@ -29,19 +29,35 @@ const WarmTierErrorMessage = ({ warmTierIndices } : Props) => {
 
   const formatTimestamp = (timestamp: number) : string => `${adjustFormat(new Date((timestamp)), 'default')}`;
 
-  const timestampInfo = warmTierIndices.map((warmTierIndex) => {
-    const begin = formatTimestamp(warmTierIndex.begin);
-    const end = formatTimestamp(warmTierIndex.end);
+  const streamsWithTimestamp = () : Array<{name: string, timestamp: number}> => {
+    const streamTimestampsList: {[key: string]: Array<number>} = {};
 
-    return `${begin} to ${end}`;
-  });
+    warmTierIndices.forEach((index) => {
+      index.stream_names.forEach((streamName) => {
+        if (!streamTimestampsList[streamName]) {
+          streamTimestampsList[streamName] = [index.begin];
+        } else {
+          streamTimestampsList[streamName].push(index.begin);
+        }
+      });
+    });
 
-  const timestampString = timestampInfo.join(', ');
+    return Object.entries(streamTimestampsList).map(([streamName, timestamps]) => {
+      const sortedTimestamps = timestamps.sort((a, b) => a - b);
+      const oldestTimestamp = sortedTimestamps[0];
+
+      return { name: streamName, timestamp: oldestTimestamp };
+    });
+  };
+
+  if (streamsWithTimestamp()?.length <= 0) return null;
```

Comment:
I think it's better to save the computed value in a variable `streamsWithTimestampMap = streamsWithTimestamp()` and reuse it where needed.